src/framework/components/collision/system.js

```diff
@@ -772,14 +776,28 @@ class CollisionComponentSystem extends ComponentSystem {
             pos = node.getPosition();
             rot = node.getRotation();
         }
-
+        const ammoQuat = new Ammo.btQuaternion();
         const transform = new Ammo.btTransform();
+
         transform.setIdentity();
         const origin = transform.getOrigin();
-        origin.setValue(pos.x, pos.y, pos.z);
+        const component = node.collision;
+
+        if (component && component._hasOffset) {
```

Comment:
Looking back on this, is this unnecessary? We should always have the component right?

Reply:
I haven't yet looked at the code, but at least here we do need a check on component. A node may come from the collision mesh implementation, where a temporary node is used, which has no collision component.